### PR TITLE
API key, secure connections, and further small changes to Elastic connection wrapper

### PR DESCRIPTION
**Pull request by filipecosta90**
This PR includes changes aimed at enhancing the Elastic connection wrapper. 
- It updates environment variables for configuration, improves the Elasticsearch client initialization with new parameters like timeout settings, and revises the handling of secure connections. 
- Additionally, it cleans up the code by removing redundant configurations and streamlining the process for setting up Elastic clients across different modules.

### CODE REVIEW

File: engine/clients/elasticsearch/config.py [hunk 1]
```diff
@@ -1,4 +1,40 @@
...
+def get_es_client(host, connection_params):
+    client: Elasticsearch = None
+    init_params = {
+        **{
+            "verify_certs": False,
+            "request_timeout": ELASTIC_TIMEOUT,
+            "retry_on_timeout": True,
+        },
+        **connection_params,
+    }
```
Comment (KShivendu) [inline engine/clients/elasticsearch/config.py:22]:
```suggestion
    init_params = {
        "verify_certs": False,
        "request_timeout": ELASTIC_TIMEOUT,
        "retry_on_timeout": True,
        **connection_params,
    }
```

File: engine/clients/elasticsearch/config.py [hunk 2]
```diff
@@ -1,4 +1,40 @@
...
+            basic_auth=(ELASTIC_USER, ELASTIC_PASSWORD),
+            **init_params,
+        )
+    else:
+        client = Elasticsearch(
+            url,
+            api_key=ELASTIC_API_KEY,
+            **init_params,
+        )
+    return client
```
Comment (KShivendu) [inline engine/clients/elasticsearch/config.py:40]:
```suggestion
    if ELASTIC_API_KEY is None:
        return Elasticsearch(
            url,
            basic_auth=(ELASTIC_USER, ELASTIC_PASSWORD),
            **init_params,
        )
    else:
        return Elasticsearch(
            url,
            api_key=ELASTIC_API_KEY,
            **init_params,
        )
```

File: engine/clients/elasticsearch/config.py [hunk 3]
```diff
@@ -1,4 +1,40 @@
...
+ELASTIC_PORT = int(os.getenv("ELASTIC_PORT", 9200))
+ELASTIC_INDEX = os.getenv("ELASTIC_INDEX", "bench")
+ELASTIC_USER = os.getenv("ELASTIC_USER", "elastic")
+ELASTIC_PASSWORD = os.getenv("ELASTIC_PASSWORD", "passwd")
+ELASTIC_API_KEY = os.getenv("ELASTIC_API_KEY", None)
+ELASTIC_TIMEOUT = int(os.getenv("ELASTIC_TIMEOUT", 90))
+
+
+def get_es_client(host, connection_params):
+    client: Elasticsearch = None
```
Comment (KShivendu) [inline engine/clients/elasticsearch/config.py:14]:
```suggestion
```

**Review by KShivendu — CHANGES_REQUESTED** [review]
Thanks again for the PR! However, I'm not sure if we want to bother about auth and API keys because we don't benchmark cloud deployments. Will think about this.

**Comment by filipecosta90** [comment]
> Thanks again for the PR! However, I'm not sure if we want to bother about auth and API keys because we don't benchmark cloud deployments. Will think about this.

@KShivendu I've removed the API key initialization. I've also extended the configurations that can be tuned and added extra checks concerning the index status and client connection. Please revise this current version

**Comment by KShivendu** [comment]
Hi @filipecosta90, I've implemented your refactor in a slightly cleaner way https://github.com/qdrant/vector-db-benchmark/pull/120

Also, I'm not sure if it makes sense to retry forcing ES into using 1 segment 30 times. Is there a simpler way to do this?

**Comment by filipecosta90** [comment]
> Also, I'm not sure if it makes sense to retry forcing ES into using 1 segment 30 times. Is there a simpler way to do this?

we can wrap it around a backoff. but lets close this PR, merge your changes and then if needed I can open a 2nd iteration PR with further changes.

**Pull request closed without merging**